### PR TITLE
Add required Content-MD5 middleware

### PR DESCRIPTION
**Pull request by gaul**
This enforces that the Content-MD5 header is present in PutObject and
UploadPart requests.

### CODE REVIEW

File: src/main/java/org/gaul/s3proxy/Main.java [hunk 1]
```diff
@@ -240,6 +240,13 @@ private static BlobStore parseMiddlewareProperties(BlobStore blobStore,
             blobStore = ReadOnlyBlobStore.newReadOnlyBlobStore(blobStore);
         }
 
+        String requiredContentMD5BlobStore = properties.getProperty(
+                S3ProxyConstants.PROPERTY_READ_ONLY_BLOBSTORE);
```
Comment (timuralp) [inline src/main/java/org/gaul/s3proxy/Main.java:244]: Should this be `PROPERTY_REQUIRED_CONTENT_MD5_BLOBSTORE`?

Comment (gaul) [reply src/main/java/org/gaul/s3proxy/Main.java]: Done.

File: src/main/java/org/gaul/s3proxy/S3ProxyHandler.java [hunk 2]
```diff
@@ -2654,6 +2657,7 @@ private void handleCopyPart(HttpServletRequest request,
                         his.hash().asBytes());
             } else {
                 Payload payload = Payloads.newInputStreamPayload(is);
+                // TODO: Content-MD5
```
Comment (gaul) [inline src/main/java/org/gaul/s3proxy/S3ProxyHandler.java:2660]: Running s3fs tests I found that this middlware cannot handle UploadPartCopy due to the emulation that S3Proxy does.  It cannot get the correct MD5 of a range of data to satisfy the underlying `RequiredContentMD5BlobStore`.

**Comment by gaul** [comment]
Closing as infeasible.

**Pull request closed without merging**